src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -234,6 +237,15 @@ struct patcher
     auto buf = bo.map<char*>();
     ofs.write(buf, bo.size());
   }
+
+  XRT_CORE_UNUSED std::string
+  generate_key_string(const std::string& argument_name, patcher::buf_type type)
+  {
+    char buf_char[16];
+    std::snprintf(buf_char, sizeof(buf_char), "%d", static_cast<int>(type));
```

Comment:
Please use modern C++ method, `std::to_string` for converting numeral types to string.